union.go

```diff
@@ -399,17 +399,22 @@ func nativeAvroFromTextualJSON(cr *codecInfo) func(buf []byte) (interface{}, []b
 			// int
 			// intNativeFromTextual
 
-			// sorted so it would be
-			// double, float, int, long
-			// that makes the priorities right by chance
-			sort.Strings(cr.allowedTypes)
+			// Sort a local copy to avoid mutating cr.allowedTypes shared state
```

Comment:
I found that union handling was mutating shared cr.allowedTypes and checkAll iterated over it instead of its parameter, which could cause wrong branch labels like "null" instead of "long".
As a result:
- Use the allowedTypes parameter inside checkAll.
- Stop sorting cr.allowedTypes and instead create a local copy to use inside checkAll